[AIRFLOW-XXXX] Remove unnecessary docstring in AWSAthenaOperator

coopergillan opened the pull request:
**EDIT - here is the updated PR description:** `query_execution_id` is returned by `execute`, which means that the default behavior is that it will be `xcom_push`ed. While this can be overridden since this is a subclass of `BaseOperator` [and this is an argument that is accepted](https://github.com/apache/airflow/blob/eb403beea2d1035635b7bea24c49b6b964313e51/airflow/models/baseoperator.py#L359), it need not be mentioned in the docstring here since it is referring to default behavior.

Old PR description for transparency:
~Since query_execution_id is now returned by execute, it will be xcom
pushed by default, meaning `do_xcom_push` is no longer needed and should
not be mentioned in the docstring.~

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

turbaszek commented on airflow/providers/amazon/aws/operators/athena.py:
```diff
@@ -28,9 +28,6 @@ class AWSAthenaOperator(BaseOperator):
     An operator that submit presto query to athena.
 
-    If ``do_xcom_push`` is True, the QueryExecutionID assigned to the
-    query will be pushed to an XCom when it successfuly completes.
```
I think this is still valid because:
https://github.com/apache/airflow/blob/7a5441836bbc8bbcb0b02bf68dbd93d63fe4ec6a/airflow/models/taskinstance.py#L1035-L1037

turbaszek replied:
If operator's `execute` method returns a `result` it doesn't mean that it will be always saved in XCom table. This will happen only if operator is used with `do_xcom_push=True`

coopergillan replied:
Got it, that makes sense. This behavior seems to correspond to the super classes and not to this specific class, hence why I don't think it needs to be mentioned in the docstring. Does that make sense?

coopergillan replied:
[Here is the setting in `BaseOperator` on `master`](https://github.com/apache/airflow/blob/eb403beea2d1035635b7bea24c49b6b964313e51/airflow/models/baseoperator.py#L359). Since operator docstrings usually do not mention super class functionality, I think this could be removed. The behavior should be evident by looking at the super class and understanding that `query_execution_id` is the value in question that will or will not be xcom_pushed.

turbaszek replied:
> Since operator docstrings usually do not mention super class functionality, I think this could be removed.

With that I can agree. My point was that this part of the PR description is not true:

>Since query_execution_id is now returned by execute, it will be xcom
pushed by default

coopergillan replied:
Noted! Just updated the PR description and title to reflect the full context. Thanks for pointing that out!

coopergillan committed:
[AIRFLOW-XXXX] Remove now-inaccurate docstring

Since query_execution_id is returned by execute, it will be xcom_pushed
by default. Since do_xcom_push is a setting in the super class,
BaseOperator, it need not be mentioned in the docstring.
